recording/lib/recording_endpoint.ex

```diff
@@ -130,7 +126,12 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording do
     track_children = track_elements ++ track_sinks
     {_track, state} = pop_in(state, [:tracks, track_id])
 
-    {[remove_children: track_children], state}
+    if state.tracks == %{} do
+      Membrane.Logger.info("All tracks were removed. Stop recording.")
+      {[remove_children: track_children, notify_parent: :finished], state}
```

Comment:
We can test it in recording integration tests.

Reply:
It is tested there. Currently, only three tests don't use that:
`empty report is not saved` because to trigger saving report without adding any tracks, we need to remove `RecordingEndpoint` forcefully to try to trigger that.
`engine doesn't crash when recording endpoint is removed too early` verifies that nothing crashes if we remove the recording before the engine handles all subscribe requests from the endpoint (it is more engine test, but it was hard for me to reproduce behavior from `RecordingEndpoint` in the engine package).
`creates correct recording, multiple inputs when recording endpoint is removed before all input endpoints finished` because it verifies what is described in the test name.
All other tests use logic selected by you.